### PR TITLE
MODINV-428: Propagate publicationPeriod property.

### PR DESCRIPTION
**Pull request by bohdan-suprun**
Propagate the `publicationPeriod` property for instance on update/create/get.
Increase `inventory` API version to `10.12`
Require `instance-storage` of `7.8` version

### CODE REVIEW

File: src/main/java/org/folio/inventory/support/JsonHelper.java [hunk 1]
```diff
@@ -28,14 +30,23 @@ public static String getString(JsonObject representation, String propertyName) {
...
     }
   }
 
+  @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+  public static <T> void includeIfPresent(
+    JsonObject representation, String propertyName, Optional<T> value) {
+
+    if (representation != null && isNotBlank(propertyName) && value.isPresent()) {
+      representation.put(propertyName, value.get());
+    }
```
Comment (hjiebsco) [inline src/main/java/org/folio/inventory/support/JsonHelper.java:47]:
```suggestion
    includeIfPresent(representation, propertyName, value.isPresent() ? value : null);
```

File: src/main/java/org/folio/inventory/support/JsonHelper.java [hunk 2]
```diff
@@ -28,14 +30,23 @@ public static String getString(JsonObject representation, String propertyName) {
...
+    JsonObject representation, String propertyName, T value) {
 
-    if (representation != null && StringUtils.isNotBlank(propertyName) && value != null) {
+    if (representation != null && isNotBlank(propertyName) && value != null) {
       representation.put(propertyName, value);
     }
   }
 
+  @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+  public static <T> void includeIfPresent(
```
Comment (hjiebsco) [inline src/main/java/org/folio/inventory/support/JsonHelper.java:42]: This method seems a bit redundant. If there is a static `PublicationPeriod.toJson(PublicationPeriod publicationPeriod)`, you can avoid adding this one.

**Review by hjiebsco — APPROVED** [review]

File: src/main/java/org/folio/inventory/support/JsonHelper.java (hunk 2 above)
Comment (bohdan-suprun) [reply src/main/java/org/folio/inventory/support/JsonHelper.java]: Make sense, I've removed this method and made the toJson to be static.

**Review by pfilippovEPAM — APPROVED** [review]
Looks good to me

**Pull request merged**